hal/api/critical.h

```diff
@@ -234,6 +287,14 @@ uint16_t core_util_atomic_incr_u16(uint16_t * valuePtr, uint16_t delta);
 uint32_t core_util_atomic_incr_u32(uint32_t * valuePtr, uint32_t delta);
 
 /**
+ * Atomic increment.
+ * @param  valuePtr Target memory location being incremented.
+ * @param  delta    The amount being incremented.
+ * @return          The new incremented value.
+ */
+void *core_util_atomic_incr(void **valuePtr, unsigned delta);
```

Comment:
I suggest you rewrite this signature using a typedef like:
`typedef void * PtrToVoid_t`, and consequentily `PtrToVoid_t core_util_atomic_incr(PtrToVoid_t *valuePtr, int pointerUnitsToIncrement);`

Please note that doing pointer arithmetic is different that doing unsigned arithmetic. The arguments to this function should specify clearly what will be incremented. In other words, if this API is used to increment the pointer by 1, then the actual value will increment by the sizeof(void *). This needs to be stated clearly.